tradeexecutor/strategy/trading_strategy_universe.py

```diff
@@ -550,4 +582,111 @@ def load_all_data(client: Client, time_frame: TimeBucket, execution_context: Exe
+    this function does a much smaller dataset download than
+    :py:func:`load_all_data`.
+
+    - This function uses optimised JSONL loading
+      via :py:meth:`~tradingstrategy.client.Client.fetch_candles_by_pair_ids`.
+
+    - Backtest data is never reloaded.
+      Furthermore, the data is stored in :py:class:`Client`
+      disk cache for the subsequent notebook and backtest runs.
+
+    - Live trading purges old data fields and reloads data
+
+    :param client:
+        Trading Strategy client instance
+
+    :param time_bucket:
+        The candle time frame
+
+    :param chain_id:
+        Which blockchain hosts our exchange
+
+    :param exchange_slug:
+        Which exchange hosts our trading pairs
+
+    :param exchange_slug:
+        Which exchange hosts our trading pairs
+
+    :param pair_tickers:
+        List of trading pair tickers as base token quote token tuples.
+        E.g. `[('WBNB', 'BUSD'), ('Cake', 'BUSD')]`.
+
+    :param liquidity:
+        Set true to load liquidity data as well
+
+    :param execution_context:
+        Defines if we are live or backtesting
+    """
+
+    assert isinstance(client, Client)
+    assert isinstance(time_bucket, TimeBucket)
+    assert isinstance(execution_context, ExecutionContext)
+    assert isinstance(chain_id, ChainId)
+    assert isinstance(exchange_slug, str)
+
+    live = execution_context.live_trading
+    with execution_context.timed_task_context_manager("load_pair_data_for_single_exchange", time_bucket=time_bucket.value):
+        if live:
+            # This will force client to redownload the data
+            logger.info("Purging trading data caches")
+            client.clear_caches()
+        else:
+            logger.info("Using cached data if available")
+
+        exchanges = client.fetch_exchange_universe()
+        pairs_df = client.fetch_pair_universe().to_pandas()
+
+        # Resolve full pd.Series for each pair
+        # we are interested in
+        our_pairs = resolve_pairs_based_on_ticker(
+            pairs_df,
+            chain_id,
+            exchange_slug,
+            pair_tickers
+        )
+
+        assert len(our_pairs) > 0, f"Pair data not found {chain_id.name}, {exchange_slug}, {pair_tickers}"
+
+        assert len(our_pairs) == len(pair_tickers), f"Pair resolution failed. Wanted to have {len(pair_tickers)} pairs, but after pair id resolution ended up with {len(our_pairs)} pairs"
+
+        our_pair_ids = set(our_pairs["pair_id"])
+
+        if len(our_pair_ids) > 1:
+            desc = f"Loading OHLCV data for {exchange_slug}"
+        else:
+            only, _ = pair_tickers
+            desc = f"Loading OHLCV data for {only[0]}-{only[1]}"
+
+        candles = client.fetch_candles_by_pair_ids(
+            our_pair_ids,
+            time_bucket,
+            progress_bar_description=desc,
+        )
+
+        if liquidity:
+            raise NotImplemented("Partial liquidity data loading is not yet supported")
```

Comment:
This will crash, it should really be `NotImplementedError`.

`NotImplemented` singleton is used as a return value in custom comparison dunder methods for signalling that an object does not know how to perform a comparison.